Build fails on Arch after introduction of explicit symbol exports for LibRegex

### Summary

Im on Arch was occasionally testing state of a project on my web apps/sites. 
But after this:  `a3754a7bf106f7d386ff9c8c1599f9da0fc2c5c1` commit it started to fail.
Im too low skilled to fix this myself but at least i reported the issue even tho you do not support Arch explicitly(as far as i know) but i suspect that the same issue might occur on other systems.
It looks like it was attempt at switching to explicit symbol exports.

P.S. Sorry for probably broken english.

### Operating system

Linux

### Steps to reproduce

1.Be on Arch
2.Checkout any commit after `a3754a7bf106f7d386ff9c8c1599f9da0fc2c5c1`
3.Run `.Meta/ladybird.sh run`


### Expected behavior

Build successfully

### Actual behavior

Build fails on LibUrl

### Log output and (if possible) backtrace


```shell
> ./Meta/ladybird.sh run
ninja: Entering directory `/home/ahito/ladybird/Build/release'
[0/2] Re-checking globbed directories...
[1/129] Linking CXX shared library lib/liblagom-url.so.0.0.0
FAILED: lib/liblagom-url.so.0.0.0
: && /sbin/c++ -fPIC -O2 -g -DNDEBUG  -fstack-protector-strong -Wl,-z,defs -Wl,--no-undefined -Wl,--no-allow-shlib-undefined -Wl,--dependency-file=Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/link.d -shared -Wl,-soname,liblagom-url.so.0 -o lib/liblagom-url.so.0.0.0 Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Host.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Origin.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Parser.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Site.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/URL.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/PublicSuffixData.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Canonicalization.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/ConstructorStringParser.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Init.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Options.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Part.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Pattern.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/PatternParser.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/String.cpp.o Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Tokenizer.cpp.o  -Wl,-rpath,/home/ahito/ladybird/Build/release/vcpkg_installed/x64-linux-dynamic/lib:/home/ahito/ladybird/Build/release/lib:  lib/liblagom-textcodec.so.0.0.0  lib/liblagom-regex.so.0.0.0  lib/liblagom-unicode.so.0.0.0  lib/liblagom-ak.so.0.0.0  -pthread  -Wl,-rpath-link,/home/ahito/ladybird/Build/release/vcpkg_installed/x64-linux-dynamic/lib && :
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `regex::Regex<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const':
/home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x3b5): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `AK::NonnullOwnPtr<regex::Regex<regex::ECMA262Parser> > AK::make<regex::Regex<regex::ECMA262Parser>, AK::ByteString, regex::RegexOptions<regex::ECMAScriptFlags>&>(AK::ByteString&&, regex::RegexOptions<regex::ECMAScriptFlags>&)':
/home/ahito/ladybird/AK/NonnullOwnPtr.h:148:(.text+0x673): undefined reference to `regex::Regex<regex::ECMA262Parser>::Regex(AK::ByteString, regex::RegexOptions<regex::ECMAScriptFlags>)'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `URL::Pattern::Component::compile(AK::Utf8View const&, AK::Function<AK::ErrorOr<AK::String, URL::Pattern::ErrorInfo> (AK::String const&)>, URL::Pattern::Options const&)':
/home/ahito/ladybird/Libraries/LibURL/Pattern/Component.cpp:248:(.text+0x6ff): undefined reference to `regex::Regex<regex::ECMA262Parser>::error_string(AK::Optional<AK::ByteString>) const'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `regex::ByteCode::~ByteCode()':
/home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196:(.text+0x8db): undefined reference to `regex::StringTable::~StringTable()'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `regex::ByteCode::~ByteCode()':
/home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196:(.text._ZN5regex8ByteCodeD2Ev[_ZN5regex8ByteCodeD5Ev]+0x1b): undefined reference to `regex::StringTable::~StringTable()'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Component.cpp.o: in function `regex::ByteCode::~ByteCode()':
/home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196:(.text._ZN5regex8ByteCodeD0Ev[_ZN5regex8ByteCodeD5Ev]+0x1b): undefined reference to `regex::StringTable::~StringTable()'
/sbin/ld: /home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196:(.text._ZN5regex5RegexINS_13ECMA262ParserEED2Ev[_ZN5regex5RegexINS_13ECMA262ParserEED5Ev]+0xfb): undefined reference to `regex::StringTable::~StringTable()'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/ConstructorStringParser.cpp.o: in function `regex::ByteCode::~ByteCode()':
/home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196:(.text+0x1516): undefined reference to `regex::StringTable::~StringTable()'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/ConstructorStringParser.cpp.o:/home/ahito/ladybird/Libraries/LibRegex/RegexByteCode.h:196: more undefined references to `regex::StringTable::~StringTable()' follow
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Pattern.cpp.o: in function `regex::Regex<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const':
/home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x74cb): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: /home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x766d): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: /home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x78fe): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: /home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x7af3): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: /home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103:(.text+0x7c6f): undefined reference to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const'
/sbin/ld: Lagom/Libraries/LibURL/CMakeFiles/LibURL.dir/Pattern/Pattern.cpp.o:/home/ahito/ladybird/Libraries/LibRegex/RegexMatcher.h:103: more undefined references to `regex::Matcher<regex::ECMA262Parser>::match(regex::RegexStringView, AK::Optional<regex::RegexOptions<regex::ECMAScriptFlags> >) const' follow
collect2: error: ld returned 1 exit status
[2/129] Building CXX object Lagom/Libraries/LibJS/CMakeFiles/LibJS.dir/Bytecode/BasicBlock.cpp.o
ninja: build stopped: subcommand failed.
```

### Screenshots or screen recordings

_No response_

### Build flags or config settings

_No response_

### Contribute a patch?

- [ ] I’ll contribute a patch for this myself.

--- a/Meta/CMake/common_options.cmake
+++ b/Meta/CMake/common_options.cmake
@@ -62,3 +62,4 @@ check_cxx_source_compiles([=[
     int main() { auto b = ^{}; auto __weak w = b; w(); }
 ]=] CXX_COMPILER_SUPPORTS_OBJC_ARC)
 unset(CMAKE_REQUIRED_FLAGS)
+unset(CMAKE_REQUIRED_LIBRARIES)

